src/Resources/config/services.yml

```diff
@@ -1,34 +1,29 @@
 services:
-    Happyr\Auth0Bundle\Factory\ConfigurationProvider:
-        arguments:
-            - ~ # config
-            - ~ # cache
-            - ~ # http client
-
-    Happyr\Auth0Bundle\Factory\Auth0Factory:
-        arguments: ['@Happyr\Auth0Bundle\Factory\ConfigurationProvider']
+    Auth0\SDK\Configuration\SdkConfiguration: ~
```

Comment:
The `SdkConfiguration` is a value object. It has getters and setters. If we define this as a service, it means that service will be stateful. This will lead to unexpected behavior. 

I prefer to keep `ConfigurationProvider` around and make sure each service get their own `SdkConfiguration` value object.

Reply:
If SdkConfiguration is not a service anymore, I could not use injection for service keys ...

Reply:
I make the SdkConfiguration as "not shared", and add a test to validate.
Seems OK